pocean/dsg/profile/im.py

```diff
@@ -166,7 +167,8 @@ def calculated_metadata(self, df=None, geometries=True, clean_cols=True, clean_r
         first_row = df.iloc[0]
         first_loc = Point(first_row.x, first_row.y)
         if geometries:
-            coords = list(unique_justseen(zip(df.x, df.y)))
+            coords = unique_justseen(zip(df.x.tolist(), df.y.tolist()))
```

Comment:
At the cost of code readability. Build for comfort, optimize when needed. You can make a PR but I'm not going to change this atm.